homeassistant/components/plugwise/climate.py

```diff
@@ -1,141 +1,153 @@
-        )
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Smile Thermostats from a config entry."""
+    api = hass.data[DOMAIN][config_entry.entry_id]["api"]
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]["coordinator"]
+
+    entities = []
+    thermostat_classes = [
+        "thermostat",
+        "zone_thermostat",
+        "thermostatic_radiator_valve",
     ]
-    add_entities(devices, True)
+    all_entities = api.get_all_devices()
+
+    for dev_id, device in all_entities.items():
+
+        if device["class"] not in thermostat_classes:
+            continue
+
+        thermostat = PwThermostat(
+            api,
+            coordinator,
+            device["name"],
+            dev_id,
+            device["location"],
+            device["class"],
+            DEFAULT_MIN_TEMP,
+            DEFAULT_MAX_TEMP,
+        )
 
+        if not thermostat:
+            continue
 
-class ThermostatDevice(ClimateEntity):
-    """Representation of the Plugwise thermostat."""
+        entities.append(thermostat)
+        _LOGGER.info("Added climate.%s", "{}".format(device["name"]))
 
-    def __init__(self, api, name, min_temp, max_temp):
+    async_add_entities(entities, True)
+
+
+class PwThermostat(SmileGateway, ClimateEntity):
+    """Representation of an Plugwise thermostat."""
+
+    def __init__(
+        self, api, coordinator, name, dev_id, loc_id, model, min_temp, max_temp
+    ):
         """Set up the Plugwise API."""
+        super().__init__(api, coordinator)
+
         self._api = api
+        self._name = name
+        self._dev_id = dev_id
+        self._loc_id = loc_id
+        self._model = model
         self._min_temp = min_temp
         self._max_temp = max_temp
-        self._name = name
-        self._direct_objects = None
-        self._domain_objects = None
-        self._outdoor_temperature = None
+
         self._selected_schema = None
         self._last_active_schema = None
         self._preset_mode = None
         self._presets = None
         self._presets_list = None
-        self._boiler_status = None
-        self._heating_status = None
-        self._cooling_status = None
-        self._dhw_status = None
+        self._boiler_state = None
+        self._heating_state = None
+        self._cooling_state = None
+        self._dhw_state = None
+        self._hvac_mode = None
         self._schema_names = None
         self._schema_status = None
-        self._current_temperature = None
-        self._thermostat_temperature = None
-        self._boiler_temperature = None
+        self._temperature = None
+        self._setpoint = None
         self._water_pressure = None
-        self._schedule_temperature = None
+        self._schedule_temp = None
         self._hvac_mode = None
+        self._single_thermostat = self._api.single_master_thermostat()
+        self._unique_id = f"cl-{dev_id}-{self._name}"
```

Comment:
Can the user chance the name of the device here?

https://developers.home-assistant.io/docs/entity_registry_index/#unique-id-requirements

Reply:
Valid point, iirc name is the suppliers name, but can adhere to sensoric name or just dev_id. @bouwew your 2cts on this?

Reply:
Discussed between ourselves, changed accordingly now